File: frontend/src/services/eventRegistrationService.js
import api from './api';

const ENDPOINT = '/event-registration';

const eventRegistrationService = {
  getAllEventRegistrations: async () => {
    try {
      const response = await api.get(`${ENDPOINT}/`);
      return response.data;
    } catch (error) {
      console.error('Error fetching event registrations:', error);
      throw error;
    }
  },

  getEventRegistrationById: async (id) => {
    try {
      const response = await api.get(`${ENDPOINT}/${id}`);
      return response.data;
    } catch (error) {
      console.error(`Error fetching event registration with id ${id}:`, error);
      throw error;
    }
  },

  createEventRegistration: async (eventRegistrationData) => {
    try {
      const response = await api.post(`${ENDPOINT}/create`, eventRegistrationData);
      return response.data;
    } catch (error) {
      console.error('Error creating event registration:', error);
      throw error;
    }
  },

  updateEventRegistration: async (id, eventRegistrationData) => {
    try {
      const response = await api.put(`${ENDPOINT}/${id}`, eventRegistrationData);
      return response.data;
    } catch (error) {
      console.error(`Error updating event registration with id ${id}:`, error);
      throw error;
    }
  },

  deleteEventRegistration: async (id) => {
    try {
      const response = await api.delete(`${ENDPOINT}/${id}`);
      return response.data;
    } catch (error) {
      console.error(`Error deleting event registration with id ${id}:`, error);
      throw error;
    }
  }
};

export default eventRegistrationService;

File: backend/src/main/java/com/skillShare/skillShare/service/EventRegistrationService.java
package com.skillShare.skillShare.service;

import com.skillShare.skillShare.dto.EventRegistrationDto;

import java.util.List;

public interface EventRegistrationService {

    // Create a new Event Registration
    EventRegistrationDto createEventRegistration(EventRegistrationDto eventRegistrationDto);

    // Get all Event Registrations
    List<EventRegistrationDto> getAllEventRegistrations();

    // Get Event Registration by ID
    EventRegistrationDto getEventRegistrationById(String id);

    // Update Event Registration by ID
    EventRegistrationDto updateEventRegistration(String id, EventRegistrationDto eventRegistrationDto);

    // Delete Event Registration by ID
    Boolean deleteEventRegistration(String id);
}
